### PR TITLE
Tl/missing documents

### PR DESCRIPTION
**Pull request by tjlav5**
Adds missing document support to Firestore with long-polling

Document ID is showing similar to console (lighter-grey, italics)
Document-preview already has "missing document copy"

### CODE REVIEW

File: src/components/Firestore/FirestoreEmulatedApiProvider.tsx [hunk 1]
```diff
@@ -149,6 +150,37 @@ export function useSubCollections(
   return collectionIds.map(id => docRef.collection(id));
 }
 
+const DOCUMENT_PATH_RE = /projects\/(?<project>.*)\/databases\/(?<database>.*)\/documents\/(?<path>.*)/;
+
+export function useMissingDocuments(
+  collection: firebase.firestore.CollectionReference
+): MissingDocument[] {
+  const { baseUrl } = useFirestoreRestApi();
+  const encodedPath = collection.path; // TODO: Encode each segment
```
Comment (yuchenshi) [inline src/components/Firestore/FirestoreEmulatedApiProvider.tsx:159]: Please encode properly -- see latest commit

File: src/components/Firestore/DocumentListItem.tsx [hunk 2]
```diff
@@ -20,37 +20,50 @@ import {
...
+import styles from './DocumentListItem.module.scss';
 import { FirestoreAny } from './models';
 import { summarize } from './utils';
 
 const DocumentListItem: React.FC<{
   docId: string;
   url: string;
   queryFieldValue: FirestoreAny | undefined;
-}> = ({ docId, url, queryFieldValue }) => {
+  hidden?: boolean;
```
Comment (yuchenshi) [inline src/components/Firestore/DocumentListItem.tsx:36]: And this property name too

**Review by yuchenshi — COMMENTED** [review]
So clean! LGTM with comments inline

File: src/components/Firestore/Collection.tsx [hunk 3]
```diff
@@ -74,6 +76,9 @@ export function withCollectionState(
     const history = useHistory();
 
     const { url } = useRouteMatch()!;
+
+    const missingDocs = useMissingDocuments(collection);
+
     // TODO: Fetch missing documents (i.e. nonexistent docs with subcollections).
```
[the remark this answers is not in the document]
Comment (tjlav5) [reply src/components/Firestore/Collection.tsx]: Removed

File: src/components/Firestore/DocumentListItem.module.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.hidden {
```
[the remark this answers is not in the document]
Comment (tjlav5) [reply src/components/Firestore/DocumentListItem.module.scss]: Changed to `missing` here and in props

File: src/components/Firestore/FirestoreEmulatedApiProvider.tsx (hunk 1 above)
Comment (tjlav5) [reply src/components/Firestore/FirestoreEmulatedApiProvider.tsx]: Pulled out into helper method.

File: src/components/Firestore/DocumentListItem.tsx (hunk 2 above)
Comment (tjlav5) [reply src/components/Firestore/DocumentListItem.tsx]: Done.

**Review by yuchenshi — APPROVED** [review]

**Pull request merged**